frontend/src/components/common/StripeCustomer.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from "react";
+import dotenv from "dotenv";
+import { loadStripe } from "@stripe/stripe-js";
+import {
+  Elements,
+  CardElement,
+  useStripe,
+  useElements,
+} from "@stripe/react-stripe-js";
+
+dotenv.config();
+
+const stripePromise = loadStripe(process.env.REACT_APP_STRIPE_PUBLISHABLE_KEY as string);
+
+const CreateCustomerForm: React.FC = () => {
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const stripe = useStripe();
+  const elements = useElements();
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!stripe || !elements) {
+      return;
+    }
+
+    const cardElement = elements.getElement(CardElement);
+
+    if (!cardElement) {
+      console.error("CardElement not found");
+      return;
+    }
+
+    const {
+      error: paymentMethodError,
+      paymentMethod,
+    } = await stripe.createPaymentMethod({
+      type: "card",
+      card: cardElement,
+      billing_details: {
+        name,
+      },
+    });
+
+    console.log("Payment Method object:", paymentMethod);
+
+    if (paymentMethodError) {
+      console.error("Error creating payment method:", paymentMethodError);
+    } else {
+      try {
+        const response = await fetch(
+          "http://localhost:5001/stripe/create-customer",
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+              name,
+              email,
+              paymentMethod: paymentMethod?.id,
+            }),
+          },
+        );
```

Comment:
can you use axios to call the backend? it keeps our codebase consistent — and the js fetch api can be clunky

Reply:
also, we should pass in the backend url from the environment variables instead of hardcoding it here

it lets us more easily change the backend url when we eventually deploy to a production environment